prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/components/Partial.jsx

```diff
@@ -37,20 +37,28 @@ export const onPartialRouteEnter = (
   { dispatch, getState },
   partialOnly = false,
   textOnly = false
-) => (nextState, replace, cb) => {
-  const postProps = propSelector(getState());
-  const { locale, moduleName } = nextState.params;
-  dispatch(setRenderPartialOnly(partialOnly));
-  dispatch(setRenderTextOnly(textOnly, { htmlTagReplacement: ' ', allowedHtmlTags: [] }));
-  dispatch(updateLocale(locale))
-    .then(() => dispatch(composeModules([{
+) => async (nextState, _replace, cb) => {
+  try {
+    const postProps = propSelector(getState());
+    const { locale, moduleName } = nextState.params;
+
+    dispatch(setRenderPartialOnly(partialOnly));
+    dispatch(setRenderTextOnly(textOnly, { htmlTagReplacement: ' ', allowedHtmlTags: [] }));
+
+    await dispatch(updateLocale(locale));
+
+    await dispatch(composeModules([{
       name: moduleName,
       props: {
         ...postProps,
         ...nextState.location.query,
       },
-    }])))
-    .then(() => cb());
+    }]));
+
+    cb();
+  } catch (error) {
+    console.error('--ERROR', error);
+  }
```

Comment:
Nothing was meaningfully changed here right? just light refactoring?

Reply:
light refactoring. The problem with react router 3 is that, if the fn throws an error, the request hangs and nothing happens. I'll revert it since this will require an extra effort to fix all the other samples and, at the end, it's just a sample